impl/src/metadata.rs

```diff
@@ -305,7 +307,8 @@ impl<'config> MetadataFetcher for CargoInternalsMetadataFetcher<'config> {
           name: target.name().to_owned(),
           kind: util::kind_to_kinds(target.kind()),
           crate_types: crate_types,
-          src_path: target.src_path().display().to_string(),
+          src_path: target.src_path().path().display().to_string(),
```

Comment:
This will panic if the target is a [metabuild](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#metabuild),  but I don't think these are yet exposed in the metadata?

Reply:
I actually wasn't aware that there was forward movement on the "metadeps" thing. I'll read into this, but I'm pretty comfortable with the change given that the tests are all passing.